design/community/book-proposal.md

```diff
@@ -0,0 +1,185 @@
+
+## Proposal
+
+Use mdbook tool to build the user-guide for the Metal³ project, which
+will combine documents under metal3-docs/docs and publish them as a single user-guide.
+
+### User Stories
+
+* As the Metal³ user I would like to have a document that would help me to
+   get started with the project easily.
+
+## Design Details
+
+### Why mdbook
+
+1. Renders a book content from markdown files
+
+2. Automatically generates all the HTML/CSS files for you
+
+3. Used by several open source projects including Kubernetes ecosystems such as
+    Cluster API, Cluster-api-provider-aws, some docs of Kubernetes
+    and etc.
+
+4. Offers preview capabilities to detect issues before they get merged.
+
+Netlify deployment will be configured for the metal3-docs repository to have a
+preview process when a future commit introduces documentation changes.
+
+**How do we keep the user-guide in sync with documentation changes in the
+    metal3-docs repository?**
+
+There are two possible ways for that.
+
+1. Have a Makefile target/script in the Metal3-docs repository to build the user-guide
+    from the current content. And configure Netlify automatic-triggering which will
+    first run the Makefile target/script under Metal3-docs and then publish the output
+    (i.e. final user-guide with updated content).
+
+2. Add the script within Netlify to build the user-guide from the current content
+    and configure Netlify automatic-triggering.
+
+**Where do we store mdbook auto-generated files like CSS/HTML ?**
+
+Since we are using Netlify, we don’t have to store those files within the Metal3-docs
+Github repository. Because, Netlify will run the mdbook binary, which will be looking
+for the book structure file in `.toml` format. As such, the user-guide is generated
+on the fly, and the output (i.e. online user-guide) is stored and published via
+the Netlify URL.
+
+**How will the book preview work when there is a PR with doc changes in the
+    Metal3-docs repository?**
+
+Netlify automatically gets triggered on every PR opened in the Metal3-docs repository.
+And after that, it will follow either step1 or step2 (in question *How do we keep
+the user-guide in sync with documentation changes in the metal3-docs repository?*)
+and publish the final-user guide and a temporary URL.
+
+**How often do we trigger the Netlify build for the final book?**
+
+Netlify allows us to configure automatic-triggering N times per day. As such,
+we can decide how many times in a day we want to trigger it. And it actually
+depends on what’s the frequency of PRs in the Metal3-docs repository. But since
+the frequency of the documentation is very low we can configure it to be once a
+day for now.
+
+### Implementation Details/Notes/Constraints
+
+Example flow of the doc changes and keeping the user-guide in sync:
+
+* The user opened a PR to change user-guide related documents.
+
+* Netlify gets triggered and publishes the user-guide based on the changes introduces
+    in the PR.
```

Comment:
Can we treat this like CI, so that if the build fails the PR won't be merged?

Reply:
Yes absolutely.